src/azimuth/state/upgrade.c

```diff
@@ -331,12 +365,19 @@ const char *az_upgrade_description(az_upgrade_t upgrade,
           return "Hold down [$o] and press [$f] to fire.";
         }
       } else if (has_multiple_rocket_ammos(upgrades)) {
-        return ("Maximum rockets increased by 5.\n"
-                "Use [9] and [0] to switch between rockets and bombs.\n"
-                "Select rockets, then hold down [$o] and press [$f] to fire.");
+        return az_show_extra_weapon_key(prefs, 9) ?
+               "Maximum rockets increased by 5.\n"
+               "Use [9]/[$9] and [0] to switch between rockets and bombs.\n"
```

Comment:
hm, i guess i don't know what you mean by the extra UI work for draw_upgrade_box_message, but here's another idea:  $0 resolves to 0 (and similarly for $k -> k) if it's not overloaded.  for this edge case, what if we don't care about the distinction for az_show_extra_weapon_key, and just always use whatever is set in the preferences?  i.e., just $0 and $9?  i know it's not showing all the ways you can set to rockets/bombs in case you have an overload, but it's showing the preference you have.  (and you can always go back and look in the menu.)  i'll push a change like this and let me know if you think that's a decent solution.

Reply:
here it is in action:
https://pasteboard.co/IR61gDV.png

Reply:
and FYI, here's a change that helps you see the upgrades (by cheating):
https://github.com/lowagner/azimuth/commit/52fbc3d63c41471e210e4457448a52cd7c52fbac
ctrl+R to get a rocket upgrade, ctrl+B to get a bombs upgrade, ctrl+G to get the guns upgrade.

Reply:
Sounds good; that does seem simpler.  (-:  The screenshot looks good to me.